src/Options.js

```diff
@@ -192,6 +192,15 @@ class Options {
         });
     }
 
+    /**
+     * Create a new options object when switching from text to math mode.
+     */
+    withMathMode(): Options {
+        return this.extend({
+            fontFamily: 'mathit',
+        });
```

Comment:
Should this also reset the fontShape and fontWeight?

Reply:
Good question. I actually thought the same thing until I broke the tests.

```
it("should render \\textsf{\\textbf{$\\mathrm{\\textsf{A}}$}} with the correct font", function() {
        const markup = katex.renderToString("\\textsf{\\textbf{$\\mathrm{\\textsf{A}}$}}");
        expect(markup).toContain("<span class=\"mord textsf textbf\">A</span>");
});
```

Basically when going from text mode, to math mode, then back to text mode, it should retain the font weight/shape from the first time it was in text mode. Hope that makes sense.

Reply:
Should it not also retain the `fontFamily`?  What happens in this case?
```
\textsf{\textbf{$\textbf{A}$}}
```
It feels like we need to:
- store font styles separate for `math` and `text` modes
- store which mode we're currently in

Reply:
Great catch. This means I'll probably resolve  #1112 in the process.